### ISSUE TITLE
Import-Package for com.googlecode.gentyref missing in vaadin-server in Vaadin 8.0.0

### ISSUE THREAD

**Issue by m1h43l** [vaadin/framework#8721]
When using com.vaadin.ui.DateField an exception is thrown because the classloader cannot load a class from the gentyref bundle because it is not stated as to be imported (in the MANIFEST.MF).

`Caused by: java.lang.NoClassDefFoundError: com/googlecode/gentyref/GenericTypeReflector
	at com.vaadin.ui.AbstractDateField.getResolutions(AbstractDateField.java:752)[218:com.vaadin.server:8.0.0]
	at com.vaadin.ui.AbstractDateField.getResolutionsHigherOrEqualTo(AbstractDateField.java:766)[218:com.vaadin.server:8.0.0]
	at com.vaadin.ui.AbstractDateField.paintContent(AbstractDateField.java:196)[218:com.vaadin.server:8.0.0]
	at com.vaadin.server.LegacyPaint.paint(LegacyPaint.java:65)[218:com.vaadin.server:8.0.0]
	at com.vaadin.server.communication.LegacyUidlWriter.write(LegacyUidlWriter.java:81)
	at com.vaadin.server.communication.UidlWriter.write(UidlWriter.java:141)
	at com.vaadin.server.communication.UIInitHandler.getInitialUidl(UIInitHandler.java:283)
	at com.vaadin.server.communication.UIInitHandler.synchronizedHandleRequest(UIInitHandler.java:80)
	at com.vaadin.server.SynchronizedRequestHandler.handleRequest(SynchronizedRequestHandler.java:41)[218:com.vaadin.server:8.0.0]
	at com.vaadin.server.VaadinService.handleRequest(VaadinService.java:1464)[218:com.vaadin.server:8.0.0]
	... 23 more`

Environment:
Vaadin 8.0.0
OSGi
Server: Karaf 4.0.8 with Jetty 9.2.19.v20160908
OpenJDK 1.8.0_111

**Comment by hesara** [comment]
Related to #8542 - the OSGi manifest for vaadin-server is missing an import for gentyref. What makes this more complicated is that the available version of gentyref is not a proper OSGi bundle.

**Comment by mpetzold** [comment]
I also had this problem. However, in my case this also came up using vaadin and gentyref embedded in an OSGi bundle. I cannot say way it can't find the class com/googlecode/gentyref/GenericTypeReflector. It works with some type of Eclipse launch configuration, but not with another. In my case the gentyref-1.2.0.jar seems to be corrupt. No idea why, maybe it does not compile as expected (I'm on Java SE 1.8).

But, I have solved the problem. I have embedded the package and classes [1] to my bundle and everything works fine now.

I would highly recommend to replace this library or create an external bundle (just as with atmosphere) for vaadin, also adding the manifest header.

[1] https://github.com/coekie/gentyref.
